Gettext last

jzigmund opened the pull request:
Last patchset contains t-keys conversion with interpolation and count variable.

Commit:
Gettext conversion - last t-keys (with count and interpolation)

Commit:
Updated PO/POT files

ifarkas commented on src/app/models/deployable.rb:
```diff
@@ -54,6 +54,14 @@ class << self
   PRESET_FILTERS_OPTIONS = []
 
+  BUILD_STATUS = {
+      :not_build => _("Images are not Built"),
```
'Built' should not be capitalized.

ifarkas commented on src/app/models/deployable.rb:
```diff
@@ -54,6 +54,14 @@ class << self
+      :not_pushed => _("Some of the images are not pushed"),
+      :pushed => _("All Images are pushed and recent.")
+  }
+
```
A problem with this solution is that the translations are being cached on the Deployable model. It means that even if you switch locale during runtime, it won't be picked up in production environment.

ifarkas commented on src/app/models/instance.rb:
```diff
@@ -120,6 +120,16 @@ class << self
+  N_('create_failed')
+  N_('error')
+  N_('vanished')
+
```
I don't like the idea of adding random Gettext function calls to the model which have nothing to do with the functionality of model. Since our models are way too oversized, how about adding these strings to locale/unfound_translations.rb or similar which is also recommended by the readme of gettext_i18n_rails?

ifarkas commented on src/app/models/deployment.rb:
```diff
@@ -502,6 +502,31 @@ def events_of_deployment_and_instances
+        _('Rollback failed, re-launch terminated')
+    end
+  end
+
```
Same here, this doesn't belong to model layer.

ifarkas commented on src/app/models/provider_account.rb:
```diff
@@ -78,6 +78,14 @@ class << self
+  N_('Account Number')
+  N_('Key')
+  N_('Certificate')
+
```
Same here, this doesn't belong to model layer.

ifarkas commented on src/app/models/user_group.rb:
```diff
@@ -26,6 +26,9 @@ class << self
 
+  N_("LDAP")
+  N_("local")
+
```
Same here, this doesn't belong to model layer.

Commit:
Move configuration of Gettext

Moved Gettext config from application.rb to initializers/fast_gettext.rb
and added default available languages.

jzigmund commented:
Pull request was updated, issues are fixed + configuration has been moved to initializers/fast_gettext.rb (according docs for fast_gettext gem)

ifarkas commented:
I think you might forget to delete the Deployable::BUILD_STATUS hash. The functionality seems redundant since it's also implemented in DeployableHelper.

jzigmund commented:
Updated and deleted mentioned hash

ifarkas commented:
I have found one more thing. In current upstream/master we can change the locale as usual:

``` ruby
FastGettext.locale # => "en" 
FastGettext.locale = :de # => :de 
FastGettext.locale # => "de" 
```

...but in your branch it is broken:

``` ruby
FastGettext.locale # => "en" 
FastGettext.locale = :de # => :de 
FastGettext.locale # => "en" 
FastGettext.locale = 'de' # => "de" 
FastGettext.locale # => "en" 
```

I am not really sure why this is happening, tried to debug it but can't find what is causing this issue. Maybe because of the changes in application.rb?

ifarkas commented:
As we discussed with Jozef, only locales listed in FastGettext.default_available_locales can be enabled with this change. It can be edited in config/initializers/fast_gettext.rb.

ACK
